### ISSUE TITLE
`exit` does nothing in scripts

### ISSUE THREAD

**Issue by donn** [RTimothyEdwards/magic#218]
```sh
#!/usr/bin/env bash
set -o pipefail

echo <<HEREDOC > test.tcl
exit -1
HEREDOC

cat /dev/null | magic -dnull -noconsole ./test.tcl
echo "Exit code: $?"


echo "exit -1" | magic -dnull -noconsole
echo "Exit code: $?"
```

While the second one does indeed print `255`, the first one simply prints `0`. Additionally, without piping /dev/null, the first one continues to offer a prompt as if nothing has happened. Is there a way around this behavior?

**Comment by donn** [comment]
This appears to extend to command failure as well: adding an unrecognized command such as `die` merely reports an error then resumes the flow.

**Comment by RTimothyEdwards** [comment]
I apologize for not seeing this when it was posted.  I just fixed it in version 8.3.447.  Mostly just a problem with vague documentation on the Tcl exit hook procedure.  It took some experimenting to establish that the clientData value passed to the exit hook is, in fact, the intended exit status.

**Comment by RTimothyEdwards** [comment]
This was tested and is known to work, so I will close out this issue.
